packages/cli/src/utils/async.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash';
+
+export async function allPromisesOrError(promisesWithObjects, toErrorMessage) {
+  const failures = []
+  const handlingFailure = async (item) => {
+    let promise, object = null
+    try {
+      if (_.isArray(item)) {
+        [promise, object] = item
+      } else {
+        promise = item
+      }
+      return await promise
+    } catch(error) {
+      failures.push([ error, object ])
+      return null
+    }
+  };
+
+  const results = await Promise.all(
+    _.map(promisesWithObjects, handlingFailure)
+  )
+
+  if(!_.isEmpty(failures)) {
```

Comment:
You can move this conditional above the result variable assignment and avoid unecessary promise management